berkdb/txn/txn_util.c

```diff
@@ -530,55 +553,64 @@ int __txn_commit_map_delete_logfile_txns(dbenv, del_log)
 	DB_ENV *dbenv;
 	u_int32_t del_log;
 {
```

Comment:
I re-wrote most of this function so I would suggest just reviewing the full function instead of the diff.